onnx/defs/object_detection/defs.cc

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "onnx/defs/schema.h"
+using namespace ONNX_NAMESPACE;
+
+namespace ONNX_NAMESPACE {
+
+static const char* RoiAlign_ver1_doc = R"DOC(
+Region of Interest (RoI) align operation described in the
+[Mask R-CNN paper](https://arxiv.org/abs/1703.06870).
+RoiAlign consumes an input tensor X and region of interests (rois)
+to apply pooling across each RoI; it produces a 4-D tensor of shape
+(num_rois, C, output_height, output_width).
+
+RoiAlign is proposed to avoid the misalignment by removing
+quantizations while converting from original image into feature
+map and from feature map into RoI feature; in each ROI bin,
+the value of the sampled locations are computed directly
+through bilinear interpolation.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    RoiAlign,
+    10,
+    OpSchema()
+      .SetDoc(RoiAlign_ver1_doc)
+      .Attr(
+          "spatial_scale",
+          "Multiplicative spatial scale factor to translate ROI coordinates "
+          "from their input spatial scale to the scale used when pooling, "
+          "i.e., spatial scale of the input feature map X relative to the "
+          "input image. E.g.; default is 1.0f. ",
+          AttributeProto::FLOAT,
+          1.f)
+      .Attr(
+          "output_height",
+          "default 1; Pooled output Y's height.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "output_width",
+          "default 1; Pooled output Y's width.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "sampling_ratio",
+          "Number of sampling points in the interpolation grid used to compute "
+          "the output value of each pooled output bin. If > 0, then exactly "
+          "sampling_ratio x sampling_ratio grid points are used. If == 0, then "
+          "an adaptive number of grid points are used (computed as "
+          "ceil(roi_width / output_width), and likewise for height). Default is 0.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Attr(
+          "mode",
+          "The pooling method. Two modes are supported: 'avg' and 'max'. "
+          "Default is 'avg'.",
+          AttributeProto::STRING,
+          std::string("avg"))
+      .Input(
+          0,
+          "X",
+          "Input data tensor from the previous operator; "
+          "4-D feature map of shape (N, C, H, W), "
+          "where N is the batch size, C is the number of channels, "
+          "and H and W are the height and the width of the data.",
+          "T1")
+      .Input(
+          1,
+          "rois",
+          "RoIs (Regions of Interest) to pool over; rois is "
```

Comment:
nit: would be good to call out that each of the co-ordinate sets has a 1-1 correspondence with the batch_index tensor. It might be obvious to some and not to others.